driver/src/examples/documentation/DocumentationSamples.java

```diff
@@ -0,0 +1,619 @@
+                Document.parse("{ item: 'journal', qty: 25, size: { h: 14, w: 21, uom: 'cm' }, status: 'A' }"),
+                Document.parse("{ item: 'notebook', qty: 50, size: { h: 8.5, w: 11, uom: 'in' }, status: 'A' }"),
+                Document.parse("{ item: 'paper', qty: 100, size: { h: 8.5, w: 11, uom: 'in' }, status: 'D' }"),
+                Document.parse("{ item: 'planner', qty: 75, size: { h: 22.85, w: 30, uom: 'cm' }, status: 'D' }"),
+                Document.parse("{ item: 'postcard', qty: 45, size: { h: 10, w: 15.25, uom: 'cm' }, status: 'A' }")
+        ));
+        // End Example 14
+
+        assertEquals(5, collection.count());
+
+        // Start Example 15
+        FindIterable<Document> findIterable = collection.find(eq("size", Document.parse("{ h: 14, w: 21, uom: 'cm' }")));
+        // End Example 15
+
+        assertEquals(1, findIterable.into(new ArrayList<Document>()).size());
+
+        // Start Example 16
+        findIterable = collection.find(eq("size", Document.parse("{ w: 21, h: 14, uom: 'cm' }")));
+        // End Example 16
+
+        assertEquals(0, findIterable.into(new ArrayList<Document>()).size());
+
+        // Start Example 17
+        findIterable = collection.find(eq("size.uom", "in"));
+        // End Example 17
+
+        assertEquals(2, findIterable.into(new ArrayList<Document>()).size());
+
+        // Start Example 18
+        findIterable = collection.find(lt("size.h", 15));
+        // End Example 18
+
+        assertEquals(4, findIterable.into(new ArrayList<Document>()).size());
+
+        // Start Example 19
+        findIterable = collection.find(and(
+                lt("size.h", 15),
+                eq("size.uom", "in"),
+                eq("status", "D")
+        ));
+        // End Example 19
+
+        assertEquals(1, findIterable.into(new ArrayList<Document>()).size());
+    }
+
+    @Test
+    public void testQueryingArrayvalues() {
+
+        //Start Example 20
+        collection.insertMany(asList(
+                Document.parse("{ item: 'journal', qty: 25, tags: ['blank', 'red'], dim_cm: [ 14, 21 ] }"),
+                Document.parse("{ item: 'notebook', qty: 50, tags: ['red', 'blank'], dim_cm: [ 14, 21 ] }"),
+                Document.parse("{ item: 'paper', qty: 100, tags: ['red', 'blank', 'plain'], dim_cm: [ 14, 21 ] }"),
+                Document.parse("{ item: 'planner', qty: 75, tags: ['blank', 'red'], dim_cm: [ 22.85, 30 ] }"),
+                Document.parse("{ item: 'postcard', qty: 45, tags: ['blue'], dim_cm: [ 10, 15.25 ] }")
+        ));
+        //End Example 20
+
+        assertEquals(5, collection.count());
+
+        //Start Example 21
+        FindIterable<Document> findIterable = collection.find(eq("tags", asList("red", "blank")));
+        //End Example 21
+
+        assertEquals(1, findIterable.into(new ArrayList<Document>()).size());
+
+        //Start Example 22
+        findIterable = collection.find(all("tags", "red", "blank"));
```

Comment:
This reads better using the overload that takes a list:

   all("tags", asList("red", "blank"))